test/src/test/java/hudson/util/BootFailureTest.java

```diff
@@ -154,6 +154,8 @@ public File allocate() throws Exception {
         @Override
         public void onLoaded() {
             wa.contextDestroyed(null);
+            // make the Jenkins.<init> thread abort
+            throw new Error();
```

Comment:
I don't think so. See https://github.com/jenkinsci/jenkins/pull/2177/commits/1f051eb37e3c33da2ed7a1d51f08ef572c4e085e where I discuss this change. `WebAppMain.contextDestroyed` does interrupt the init thread, but the problem is that the init thread isn't waiting by this point in `Jenkins.<init>` that invokes this callback. So the interrupt will never get processed.

This test case from get-go incorrectly relied on the presence of the `Object.wait()` code in the init thread after this callback, which is no longer true.